src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -290,6 +291,80 @@ class PersonAgent(
 
   override def logDepth: Int = 30
 
+  val distanceSafetyMargin: Double = beamScenario.beamConfig.beam.agentsim.agents.parking.bev.distance_safety_margin
+
+  /**
+    * identifies agents with remaining range which is smaller than their remaining tour
+    *
+    * @param personData current state data cast as a [[BasePersonData]]
+    * @return true if they have enough fuel, or fuel type is not exhaustible
+    */
+  def calculateRemainingTripData(personData: BasePersonData): Option[ParkingMNL.RemainingTripData] = {
+
+    val refuelNeeded: Boolean =
+      if (currentBeamVehicle.isCAV) {
+        // not human, use non-human config values
+        currentBeamVehicle.isRefuelNeeded(
+          beamScenario.beamConfig.beam.agentsim.agents.rideHail.cav.refuelRequiredThresholdInMeters,
+          beamScenario.beamConfig.beam.agentsim.agents.rideHail.cav.noRefuelThresholdInMeters
+        )
+      } else {
+
+        currentBeamVehicle.isRefuelNeeded(
+          beamScenario.beamConfig.beam.agentsim.agents.rideHail.human.refuelRequiredThresholdInMeters,
```

Comment:
Using rideHail parameters to decide whether a personal EV driver needs to charge? That's likely to going to cause problems.

Reply:
ok, that makes sense. i just figured CAVs needed to be handled differently. those lines don't need to be there, so i will remove them.